src/ldp/auxiliary/RoutingAuxiliaryIdentifierStrategy.ts

```diff
@@ -0,0 +1,44 @@
+import { InternalServerError } from '../../util/errors/InternalServerError';
+import { NotImplementedHttpError } from '../../util/errors/NotImplementedHttpError';
+import type { ResourceIdentifier } from '../representation/ResourceIdentifier';
+import type { AuxiliaryIdentifierStrategy } from './AuxiliaryIdentifierStrategy';
+
+/**
+ * An {@link AuxiliaryIdentifierStrategy} that combines multiple AuxiliaryIdentifierStrategies into one.
+ * Uses `isAuxiliaryIdentifier` to know which strategy to route to.
+ */
+export class RoutingAuxiliaryIdentifierStrategy implements AuxiliaryIdentifierStrategy {
+  protected readonly sources: AuxiliaryIdentifierStrategy[];
+
+  public constructor(sources: AuxiliaryIdentifierStrategy[]) {
+    this.sources = sources;
+  }
+
+  public getAuxiliaryIdentifier(): never {
```

Comment:
Is the singular method ever used? Should we remove it?

Reply:
It is used by classes that need a specific auxiliary manager. There are some classes (acl authorizer, link metadata generator) that specifically need an acl based strategy, and those then use that function. They could also use the other one and then take the first element of the array of course, but that seems less nice.